aiida/orm/implementation/django/logs.py

```diff
@@ -95,24 +97,59 @@ def delete(self, log_id):
         """
         Remove a Log entry from the collection with the given id
 
-        :param log_id: id of the log to delete
+        :param log_id: id of the Log to delete
+        :type log_id: int
+
+        :raises TypeError: if ``log_id`` is not an `int`
+        :raises `~aiida.common.exceptions.NotExistent`: if Log with ID ``log_id`` is not found
         """
-        # pylint: disable=no-name-in-module,import-error
-        from django.core.exceptions import ObjectDoesNotExist
-        assert log_id is not None
+        if not isinstance(log_id, int):
+            raise TypeError("log_id must be an int")
+
         try:
             models.DbLog.objects.get(id=log_id).delete()
         except ObjectDoesNotExist:
             raise exceptions.NotExistent("Log with id '{}' not found".format(log_id))
 
+    def delete_all(self):
+        """
+        Delete all Log entries.
+
+        :raises `~aiida.common.exceptions.IntegrityError`: if all Logs could not be deleted
+        """
+        from django.db import transaction
+        try:
+            with transaction.atomic():
+                models.DbLog.objects.all().delete()
+        except Exception as exc:
```

Comment:
surprised pylint doesn't throw a fit here about the broad-exception. This file is not excluded from pre-commit no?

Reply:
Nope - pylint just knows about my awesomeness, and cannot bear to complain :smile: